mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -446,6 +463,20 @@ private[ml] object DefaultParamsReader {
     val cls = Utils.classForName(metadata.className)
     cls.getMethod("read").invoke(null).asInstanceOf[MLReader[T]].load(path)
   }
+
+  def loadAndSetInitialModel[M <: Model[M]](
+      instance: HasInitialModel[M], metadata: Metadata, path: String, sc: SparkContext): Unit = {
+    implicit val format = DefaultFormats
+    // Try to load the initial model
```

Comment:
What about just wrapping the load initial model in a `Try`? That way there is no need to rely on metadata. It will fail if the "initialModel" path doesn't exist. We can ignore that exception but throw (or log) if we encounter an error in the actual loading.